gio/src/unix_fd_list.rs

```diff
@@ -42,39 +42,41 @@ pub trait UnixFDListExtManual: IsA<UnixFDList> + Sized {
     }
 
     #[doc(alias = "g_unix_fd_list_get")]
-    fn get(&self, index_: i32) -> Result<RawFd, glib::Error> {
+    fn get(&self, index_: i32) -> Result<OwnedFd, glib::Error> {
         unsafe {
             let mut error = ptr::null_mut();
-            let ret = ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let raw_fd =
+                ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let fd = OwnedFd::from_raw_fd(raw_fd);
             if error.is_null() {
-                Ok(ret)
+                Ok(fd)
             } else {
                 Err(from_glib_full(error))
             }
         }
     }
 
     #[doc(alias = "g_unix_fd_list_peek_fds")]
-    fn peek_fds(&self) -> Vec<RawFd> {
+    fn peek_fds(&self) -> &[BorrowedFd<'_>] {
         unsafe {
             let mut length = mem::MaybeUninit::uninit();
-            let ret = FromGlibContainer::from_glib_none_num(
-                ffi::g_unix_fd_list_peek_fds(self.as_ref().to_glib_none().0, length.as_mut_ptr()),
+            std::slice::from_raw_parts(
+                ffi::g_unix_fd_list_peek_fds(self.as_ref().to_glib_none().0, length.as_mut_ptr())
+                    as *const BorrowedFd,
                 length.assume_init() as usize,
-            );
-            ret
+            )
         }
     }
 
     #[doc(alias = "g_unix_fd_list_steal_fds")]
-    fn steal_fds(&self) -> Vec<RawFd> {
+    fn steal_fds(&self) -> Vec<OwnedFd> {
```

Comment:
I think this invalidates the return value of `peek_fds()`, so you'd be left with a dangling pointer

Reply:
Would it work to make peek_fds return a RawFd (again) and mark it as unsafe?

Reply:
Doesn't have to be `unsafe` but yes, returning `Vec<RawFd>` seems like the only sensible choice here

Reply:
Done.